src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,84 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.InvalidAgeException;
+import core.basesyntax.exception.LoginValidateException;
+import core.basesyntax.exception.PasswordValidateException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
```

Comment:
still missing test cases that we should add:
* happy case - when user is valid, no validation throws exception, and it gets put in the storage
* null login/age/password - asserts the corresponding exception is thrown
* when trying to register a user, and we already have a user with the same login in the storage - also asserts that exception is thrown. P.S. Definitely check the common mistakes for details on how to put the first user in the `Storage` properly!